### PR TITLE
Stratified report fix

### PR DESCRIPTION
**Pull request by lmd59**
# Summary
Addresses UnexpectedProperty issue in https://github.com/projecttacoma/fqm-execution/issues/311. 

## New behavior
For episode-based measures with stratifiers, (as was already done for patient-based), we now default the `strataCode` to be the id if the `strata.code?.text` does not exist

## Code changes

- `DetailedResultsBuilder.ts` - defaults `strataCode` to `strata.id` as secondary preferred value for episode results
- `MeasureReportBuilder.ts` - slight efficiency improvement and added clarifying comment
- Adds tests addressing episode+stratifier cases in `DetailedResultsBuilder.test.ts` an `MeasureReportBuilder.test.ts`

# Testing guidance

- `npm run check`
- Pull measure/patient from ticket and test using `fqm-execution details -m "path\to\measures\EpisodeMeasure.json" -p "path\to\patients\Episode_Patient.json" -o "path\to\output\Episode_Details.json" --vs-api-key {apiKey}`

### CODE REVIEW

File: src/calculation/MeasureReportBuilder.ts [hunk 1]
```diff
@@ -647,9 +647,9 @@ export default class MeasureReportBuilder<T extends PopulationGroupResult> exten
     options: CalculationOptions
   ): fhir4.MeasureReport[] {
     const reports: fhir4.MeasureReport[] = [];
+    const measure = extractMeasureFromBundle(measureBundle);
+    const builder = new MeasureReportBuilder(measure, options);
```
Comment (hossenlopp) [inline src/calculation/MeasureReportBuilder.ts:651]: This line needs to be inside the `forEach` loop below as it needs to be creating a new MeasureReport for each patient.

Good call to move the `extractMeasureFromBundle` outside the loop.

**Review by hossenlopp — CHANGES_REQUESTED** [review]

File: src/calculation/MeasureReportBuilder.ts (hunk 1 above)
Comment (lmd59) [reply src/calculation/MeasureReportBuilder.ts:651]: whoops! Thank you for the catch!

File: test/unit/MeasureReportBuilder.test.ts [hunk 2]
```diff
@@ -490,6 +606,49 @@ describe('MeasureReportBuilder Static', () => {
...
+    let measureReports: fhir4.MeasureReport[];
+    beforeAll(() => {
+      measureReports = MeasureReportBuilder.buildMeasureReports(
+        episodeWithStratMeasureBundle,
+        episodeWithStratExecutionResults,
+        calculationOptions
+      );
+    });
+
+    test('should generate 1 result', () => {
```
Comment (elsaperelli) [inline test/unit/MeasureReportBuilder.test.ts:619]: Maybe be a bit more specific here like "Generates one measure report"

**Review by elsaperelli — APPROVED** [review]
🥳

**Pull request merged**